app/src/main/java/com/checkin/app/checkin/Home/NearbyRestaurantAdapter.kt

```diff
@@ -1,43 +1,126 @@
+import java.net.URL
+import java.util.*
+import android.graphics.BitmapFactory
+import android.graphics.Bitmap
+import android.nfc.Tag
+
+
+class NearbyRestaurantAdapter(var context: Context) : RecyclerView.Adapter<BaseViewHolder<Any>>() {
+
+
 
-class NearbyRestaurantAdapter : RecyclerView.Adapter<BaseViewHolder<Any>>() {
     var data: List<NearbyRestaurantModel> = emptyList()
 
     fun updateData(newData: List<NearbyRestaurantModel>) {
+
         data = newData
         notifyDataSetChanged()
     }
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder<Any> {
         val view = LayoutInflater.from(parent.context).inflate(viewType, parent, false)
-        return if (viewType == R.layout.item_home_restaurant_banner) NearbyRestaurantViewHolder(view) as BaseViewHolder<Any> else AdvertisementViewHolder(view)
+        return if (viewType == R.layout.item_home_restaurant_banner) NearbyRestaurantViewHolder(view)
+                as BaseViewHolder<Any> else AdvertisementViewHolder(view)
     }
 
-    override fun getItemCount(): Int = data.count() + 1
+    override fun getItemCount(): Int = data.count()
 
-    override fun getItemViewType(position: Int): Int = if (position == 2) R.layout.item_advertisement_banner else R.layout.item_home_restaurant_banner
+    override fun getItemViewType(position: Int): Int = if (position == 2)
+        R.layout.item_advertisement_banner else R.layout.item_home_restaurant_banner
 
     override fun onBindViewHolder(holder: BaseViewHolder<Any>, position: Int) {
         holder.bindData(data[position])
     }
 
     inner class NearbyRestaurantViewHolder(itemView: View) : BaseViewHolder<NearbyRestaurantModel>(itemView) {
         override fun bindData(data: NearbyRestaurantModel) {
-            TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+            var imageView:ImageView=itemView.findViewById(R.id.home_image)
+            imageView.setImageResource(data.url)
         }
     }
 
     inner class AdvertisementViewHolder(itemView: View) : BaseViewHolder<Any>(itemView) {
+
         override fun bindData(data: Any) {
-            TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+             var vpBanner: ViewPager=itemView.findViewById(R.id.vp_home_banner)
+             var mPagerAdapter: BannerPagerAdapter=BannerPagerAdapter(context);
+            var indicatorView: PageIndicatorView? = itemView.findViewById(R.id.indicator_home_banner)
+
+            vpBanner.adapter = mPagerAdapter
+            if(indicatorView!=null) {
+                indicatorView.setViewPager(vpBanner)
+
+                indicatorView.setAnimationType(AnimationType.FILL)
+                indicatorView.setClickListener { position -> vpBanner.currentItem = position }
+            }
+            else{
+
+            }
+
+            vpBanner.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
```

Comment:
Just like I explained above, this doesn't deal with data so can be moved to constructor of the ViewHolder